libs/platform/ebpf_hash_table.c

```diff
@@ -108,10 +108,7 @@ ebpf_hash_table_destroy(_Pre_maybenull_ _Post_invalid_ ebpf_hash_table_t* hash_t
 }
 
 ebpf_result_t
-ebpf_hash_table_find(
-    _In_ ebpf_hash_table_t* hash_table,
-    _In_ _Pre_readable_byte_size_(hash_table->key_size) const uint8_t* key,
-    _Outptr_ _Post_readable_size_(hash_table->value_size) uint8_t** value)
+ebpf_hash_table_find(_In_ ebpf_hash_table_t* hash_table, _In_ const uint8_t* key, _Outptr_ uint8_t** value)
```

Comment:
Doesn't this trigger warnings assuming the key is 1 byte long?

Reply:
Nope.  That's because the key is passed as the second argument to the following, where the SAL annotation does not expect anything:
```
_Must_inspect_result_
NTSYSAPI
PVOID
NTAPI
RtlLookupElementGenericTableAvl (
    _In_ PRTL_AVL_TABLE Table,
    _In_ PVOID Buffer
    );
```

Reply:
But the Rtl API will eventually invoke _ebpf_hash_map_compare which does look at key size. But I doubt if the static analyzer is that smart to hunt through function pointers.

Reply:
_ebpf_hash_map_compare isn't yet SAL annotated at all it seems.